### PR TITLE
Check RDT Flags Before Getting Frame's Doc Data

### PR DESCRIPTION
**Pull request by RadwanFaci**
Adding a check to see if a document's RDT flags indicate that it has been initialized. We should only request the doc data if it's been initialized to avoid blocking the UI thread.

### CODE REVIEW

**Comment by RadwanFaci** [comment]
@dotnet-policy-service agree company="Microsoft"

**Review by sharwell — CHANGES_REQUESTED** [review]
I'm reviewing in detail to see if we can simplify this

File: src/VisualStudio/Core/Def/Workspace/VisualStudioActiveDocumentTracker.cs [hunk 1]
```diff
@@ -230,7 +234,14 @@ public FrameListener(VisualStudioActiveDocumentTracker service, IVsWindowFrame f
...
 
-                if (ErrorHandler.Succeeded(frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocData, out var docData)))
+                var isDocDataInitialized = false;
+                if (ErrorHandler.Succeeded(frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocCookie, out var docCookieProp)) && docCookieProp is uint docCookie)
+                {
+                    RunningDocumentInfo runningDocumentInfo = new(service._runningDocumentTable, docCookie);
+                    isDocDataInitialized = runningDocumentInfo.IsDocumentInitialized;
+                }
+
+                if (isDocDataInitialized && ErrorHandler.Succeeded(frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocData, out var docData)))
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Workspace/VisualStudioActiveDocumentTracker.cs:244]: 📝 I'm not having success here. This is going to leave `TextBuffer` permanently uninitialized, and we need this value elsewhere in code.

**Comment by CyrusNajmabadi** [comment]
i agree with Sam.  If this synchronously blocks the UI, and we want to avoid that, what's the non-blocking alternative to allow us to still actually get the buffer?  @olegtk ?

**Pull request closed without merging**